src/main/resources/org/broadinstitute/dsde/sam/liquibase/changesets/20240417_action_managed_identities.xml

```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog logicalFilePath="dummy"
+                   xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
+
+    <changeSet logicalFilePath="dummy" author="tlangs" id="action_managed_identities">
+        <createTable tableName="SAM_ACTION_MANAGED_IDENTITY">
```

Comment:
Should this table have any unique constraints? Like should an `action_managed_identity` be unique by (resource, action, mrg)?

Also -- I know we're starting with org-wide identities for now. If we discover we want a per-workspace or per-user/workspace action managed identity in the future, how extensible is this? (Not saying we necessarily need to engineer for this now, just curious.)

Reply:
The primary key being the `resource_id`, `resource_action_id`, and `azure_managed_resource_group_id` will make sure they're unique.

Action Managed Identities can be applied to any action on any resource, so they're very extensible in that way. We don't have the capacity to do per-user Action Managed Identities, but that also kinda flies in the face of what Action Managed Identities are meant to be used for. Namely, they're a way to abstract over Terra users and represent the Terra platform's access to something.